modules/utils/src/merkle.ts

```diff
@@ -9,13 +9,17 @@ import { encodeCoreTransferState, hashCoreTransferState } from "./transfers";
 export const generateMerkleTreeData = (transfers: CoreTransferState[]): { root: string; tree: merkle.Tree } => {
   // Create leaves
   const tree = new merkle.Tree();
-  tree.free(); // handle memory leaks
   transfers.forEach((transfer) => {
     tree.insert_hex_js(encodeCoreTransferState(transfer));
   });
 
   // Return
-  const calculated = tree.root_js();
+  let calculated: string;
+  try {
+    calculated = tree.root_js();
```

Comment:
The part where the transfers are inserted needs to be in the body of the `try` block.